src/ResourceManager/EventGrid/Commands.EventGrid.Test/ScenarioTests/EventSubscriptionTests.ps1

```diff
@@ -400,4 +485,32 @@ function EventSubscriptionTests_Subscription2 {
 
     Write-Debug " Deleting event subscription: $eventSubscriptionName2"
     Remove-AzureRmEventGridSubscription -EventSubscriptionName $eventSubscriptionName2
-}
\ No newline at end of file
+}
+
+<#
+.SYNOPSIS
+Tests EventGrid EventSubscription with deadletter destination
+#>
+function EventSubscriptionTests_Deadletter {
+    # Setup
+    $subscriptionId = Get-SubscriptionId
+    $location = Get-LocationForEventGrid
+    $eventSubscriptionName = Get-EventSubscriptionName
+    $eventSubscriptionEndpoint = "https://eventgridrunnerfunction.azurewebsites.net/api/HttpTriggerCSharp1?code=<HIDDEN>"
+    $deadletterResourceId = "/subscriptions/$subscriptionId/resourceGroups/testpowershelResourceGroup/providers/Microsoft.Storage/storageAccounts/testpsstorageaccount/blobServices/default/containers/psdeadletterqueue"
+
+    try
+    {
+        Write-Debug " Creating a new EventSubscription $eventSubscriptionName to Azure subscription"
+        $labels = "Finance", "HR"
+        $result = New-AzureRmEventGridSubscription -Endpoint $eventSubscriptionEndpoint -EventSubscriptionName $eventSubscriptionName -Label $labels -DeadLetterEndpoint $deadletterResourceId
+        Assert-True {$false} "New-AzureRmEventGridSubscription succeeded while it is expected to fail as deadletter is not yet enabled"
```

Comment:
Why are you adding a parameter if including it will always cause the cmdlet to fail? It seems like this should be implemented in a future iteration.

Reply:
This feature is completed and part of our swagger and CLI. We disabled it temporary to finalize few bug fixes which are almost done and deployed. So, we would need to keep this as part of the powershell too.

Reply:
We either need to remove this parameter, or wait until the fix is deployed and the test is updated